### ISSUE TITLE
Various exceptions when entering multi-root workspace

### ISSUE THREAD

**Issue by bpasero** [Microsoft/vscode#48065]
Steps to Reproduce:

1. have a folder "`foo`" with 2 children folder "`a`" and "`b`"
2. open folder "`foo`"
3. drag and drop the folders "`a`" and "`b`" from outside into the explorer
4. confirm to add to the workspace

Watch the dev console spam:

```
[Extension Host] debugger listening on port 5870
/Users/bpasero/Development/Microsoft/monaco/out/vs/workbench/services/extensions/electron-browser/extensionHost.js:145 [Extension Host] debugger inspector at chrome-devtools://devtools/bundled/inspector.html?experiments=true&v8only=true&ws=127.0.0.1:5870/06c72ab8-efab-43a2-b588-aa4d6756aff0
log.js:200   ERR Cannot read property 'activateByEvent' of null: TypeError: Cannot read property 'activateByEvent' of null
    at ExtensionService._activateByEvent (file:///Users/bpasero/Development/Microsoft/monaco/out/vs/workbench/services/extensions/electron-browser/extensionService.js:364:54)
    at ExtensionService.activateByEvent (file:///Users/bpasero/Development/Microsoft/monaco/out/vs/workbench/services/extensions/electron-browser/extensionService.js:354:29)
    at file:///Users/bpasero/Development/Microsoft/monaco/out/vs/workbench/services/mode/common/workbenchModeService.js:125:41
    at Emitter.fire (file:///Users/bpasero/Development/Microsoft/monaco/out/vs/base/common/event.js:103:38)
    at WorkbenchModeServiceImpl.ModeServiceImpl._getOrCreateMode (file:///Users/bpasero/Development/Microsoft/monaco/out/vs/editor/common/services/modeServiceImpl.js:118:39)
    at file:///Users/bpasero/Development/Microsoft/monaco/out/vs/editor/common/services/modeServiceImpl.js:111:30
    at Promise_ctor.CompletePromise_then [as then] (file:///Users/bpasero/Development/Microsoft/monaco/out/vs/base/common/winjs.base.js:1581:49)
    at WorkbenchModeServiceImpl.ModeServiceImpl.getOrCreateModeByFilenameOrFirstLine (file:///Users/bpasero/Development/Microsoft/monaco/out/vs/editor/common/services/modeServiceImpl.js:108:36)
    at TextFileEditorModel.getOrCreateMode (file:///Users/bpasero/Development/Microsoft/monaco/out/vs/workbench/services/textfile/common/textFileEditorModel.js:409:32)
    at TextFileEditorModel.BaseTextEditorModel.createTextEditorModel (file:///Users/bpasero/Development/Microsoft/monaco/out/vs/workbench/common/editor/textEditorModel.js:73:29)
    at file:///Users/bpasero/Development/Microsoft/monaco/out/vs/workbench/services/textfile/common/textFileEditorModel.js:369:30
    at CompletePromise_ctor.CompletePromise_then [as then] (file:///Users/bpasero/Development/Microsoft/monaco/out/vs/base/common/winjs.base.js:1581:49)
    at TextFileEditorModel.doCreateTextModel (file:///Users/bpasero/Development/Microsoft/monaco/out/vs/workbench/services/textfile/common/textFileEditorModel.js:367:75)
    at TextFileEditorModel.doLoadWithContent (file:///Users/bpasero/Development/Microsoft/monaco/out/vs/workbench/services/textfile/common/textFileEditorModel.js:346:25)
    at TextFileEditorModel.loadWithContent (file:///Users/bpasero/Development/Microsoft/monaco/out/vs/workbench/services/textfile/common/textFileEditorModel.js:292:25)
    at TextFileEditorModel.handleLoadSuccess (file:///Users/bpasero/Development/Microsoft/monaco/out/vs/workbench/services/textfile/common/textFileEditorModel.js:270:25)
    at file:///Users/bpasero/Development/Microsoft/monaco/out/vs/workbench/services/textfile/common/textFileEditorModel.js:265:57
    at Object.notifySuccess [as _notify] (file:///Users/bpasero/Development/Microsoft/monaco/out/vs/base/common/winjs.base.js:1185:59)
    at Object.enter (file:///Users/bpasero/Development/Microsoft/monaco/out/vs/base/common/winjs.base.js:867:30)
    at Promise_ctor._run (file:///Users/bpasero/Development/Microsoft/monaco/out/vs/base/common/winjs.base.js:1083:29)
    at Promise_ctor._completed (file:///Users/bpasero/Development/Microsoft/monaco/out/vs/base/common/winjs.base.js:1051:18)
    at IconvLiteDecoderStream.<anonymous> (file:///Users/bpasero/Development/Microsoft/monaco/out/vs/editor/common/model/textModel.js:46:21)
    at emitNone (events.js:111:20)
    at IconvLiteDecoderStream.emit (events.js:208:7)
    at endReadableNT (_stream_readable.js:1056:12)
    at _combinedTickCallback (internal/process/next_tick.js:138:11)
    at process._tickCallback (internal/process/next_tick.js:180:9)
extensionHost.js:230 [Extension Host] Provided debugging port 5870 is not free, using 5871 instead.
(anonymous) @ extensionHost.js:230
_tickCallback @ internal/process/next_tick.js:188
Promise resolved (async)
(anonymous) @ extensionHost.js:224
Promise_ctor @ winjs.base.js:1656
ExtensionHostProcessWorker._tryFindDebugPort @ extensionHost.js:223
ExtensionHostProcessWorker.start @ extensionHost.js:93
ExtensionHostProcessManager @ extensionService.js:128
create @ types.js:164
InstantiationService._createInstance @ instantiationService.js:106
InstantiationService.createInstance @ instantiationService.js:67
ExtensionService._startExtensionHostProcess @ extensionService.js:320
ExtensionService.startExtensionHost @ extensionService.js:300
startExtensionHost @ workspaceEditingService.js:152
notifySuccess @ winjs.base.js:1185
enter @ winjs.base.js:867
_run @ winjs.base.js:1083
_completed @ winjs.base.js:1051
argDone @ winjs.base.js:1828
Promise.then.errors.(anonymous function) @ winjs.base.js:1853
notifySuccess @ winjs.base.js:1185
enter @ winjs.base.js:867
_run @ winjs.base.js:1083
_completed @ winjs.base.js:1051
CompletePromise_then @ winjs.base.js:1581
enter @ winjs.base.js:762
_run @ winjs.base.js:1083
_setCompleteValue @ winjs.base.js:1067
notifySuccess @ winjs.base.js:1185
enter @ winjs.base.js:916
_run @ winjs.base.js:1083
_error @ winjs.base.js:1056
(anonymous) @ async.js:602
(anonymous) @ extfs.js:37
(anonymous) @ /Users/bpasero/Development/Microsoft/monaco/node_modules/graceful-fs/polyfills.js:284
(anonymous) @ original-fs.js:152
extensionHost.js:236 [Extension Host] debugger listening on port 5871
extensionHost.js:145 [Extension Host] debugger inspector at chrome-devtools://devtools/bundled/inspector.html?experiments=true&v8only=true&ws=127.0.0.1:5871/b6bf586c-d30a-4e62-8253-5b317d05f5b9
log.js:200   ERR Reviver for 'mainThreadWebview' already registered: Error: Reviver for 'mainThreadWebview' already registered
    at WebviewEditorService.registerReviver (file:///Users/bpasero/Development/Microsoft/monaco/out/vs/workbench/parts/webview/electron-browser/webviewEditorService.js:105:23)
    at MainThreadWebviews (file:///Users/bpasero/Development/Microsoft/monaco/out/vs/workbench/api/electron-browser/mainThreadWebview.js:61:29)
    at create (file:///Users/bpasero/Development/Microsoft/monaco/out/vs/base/common/types.js:164:14)
    at InstantiationService._createInstance (file:///Users/bpasero/Development/Microsoft/monaco/out/vs/platform/instantiation/common/instantiationService.js:106:35)
    at InstantiationService.createInstance (file:///Users/bpasero/Development/Microsoft/monaco/out/vs/platform/instantiation/common/instantiationService.js:67:29)
    at ExtensionHostProcessManager._createExtensionHostCustomers (file:///Users/bpasero/Development/Microsoft/monaco/out/vs/workbench/services/extensions/electron-browser/extensionService.js:180:59)
    at file:///Users/bpasero/Development/Microsoft/monaco/out/vs/workbench/services/extensions/electron-browser/extensionService.js:129:39
    at Object.notifySuccess [as _notify] (file:///Users/bpasero/Development/Microsoft/monaco/out/vs/base/common/winjs.base.js:1185:59)
    at Object.enter (file:///Users/bpasero/Development/Microsoft/monaco/out/vs/base/common/winjs.base.js:867:30)
    at _Base.Class.derive._creator._run (file:///Users/bpasero/Development/Microsoft/monaco/out/vs/base/common/winjs.base.js:1083:29)
    at _Base.Class.derive._creator._completed (file:///Users/bpasero/Development/Microsoft/monaco/out/vs/base/common/winjs.base.js:1051:18)
    at Object.notifySuccess [as _notify] (file:///Users/bpasero/Development/Microsoft/monaco/out/vs/base/common/winjs.base.js:1185:59)
    at Object.enter (file:///Users/bpasero/Development/Microsoft/monaco/out/vs/base/common/winjs.base.js:867:30)
    at Promise_ctor._run (file:///Users/bpasero/Development/Microsoft/monaco/out/vs/base/common/winjs.base.js:1083:29)
    at Promise_ctor._completed (file:///Users/bpasero/Development/Microsoft/monaco/out/vs/base/common/winjs.base.js:1051:18)
    at file:///Users/bpasero/Development/Microsoft/monaco/out/vs/workbench/services/extensions/electron-browser/extensionHost.js:279:29
    at Emitter.fire (file:///Users/bpasero/Development/Microsoft/monaco/out/vs/base/common/event.js:103:38)
    at Socket.<anonymous> (file:///Users/bpasero/Development/Microsoft/monaco/out/vs/base/parts/ipc/node/ipc.net.js:90:46)
    at emitOne (events.js:116:13)
    at Socket.emit (events.js:211:7)
    at addChunk (_stream_readable.js:263:12)
    at readableAddChunk (_stream_readable.js:250:11)
    at Socket.Readable.push (_stream_readable.js:208:10)
    at Pipe.onread (net.js:594:20)
errors.js:42 WARNING: Promise with no error callback:36
errors.js:43 {exception: Error: Reviver for 'mainThreadWebview' already registered
    at WebviewEditorService.registerRevive…, error: null, promise: _…e.C…s.d…e._creator, handler: undefined, id: 36, …}
errors.js:45 Error: Reviver for 'mainThreadWebview' already registered
    at WebviewEditorService.registerReviver (webviewEditorService.js:105)
    at MainThreadWebviews (mainThreadWebview.js:61)
    at create (types.js:164)
    at InstantiationService._createInstance (instantiationService.js:106)
    at InstantiationService.createInstance (instantiationService.js:67)
    at ExtensionHostProcessManager._createExtensionHostCustomers (extensionService.js:180)
    at extensionService.js:129
    at Object.notifySuccess [as _notify] (winjs.base.js:1185)
    at Object.enter (winjs.base.js:867)
    at _Base.Class.derive._creator._run (winjs.base.js:1083)
    at _Base.Class.derive._creator._completed (winjs.base.js:1051)
    at Object.notifySuccess [as _notify] (winjs.base.js:1185)
    at Object.enter (winjs.base.js:867)
    at Promise_ctor._run (winjs.base.js:1083)
    at Promise_ctor._completed (winjs.base.js:1051)
    at extensionHost.js:279
    at Emitter.fire (event.js:103)
    at Socket.<anonymous> (ipc.net.js:90)
    at emitOne (events.js:116)
    at Socket.emit (events.js:211)
    at addChunk (_stream_readable.js:263)
    at readableAddChunk (_stream_readable.js:250)
    at Socket.Readable.push (_stream_readable.js:208)
    at Pipe.onread (net.js:594)
log.js:200   ERR Unknown actor mMainThreadTextEditors: Error: Unknown actor mMainThreadTextEditors
    at RPCProtocol._doInvokeHandler (file:///Users/bpasero/Development/Microsoft/monaco/out/vs/workbench/services/extensions/node/rpcProtocol.js:138:23)
    at RPCProtocol._invokeHandler (file:///Users/bpasero/Development/Microsoft/monaco/out/vs/workbench/services/extensions/node/rpcProtocol.js:130:54)
    at RPCProtocol._receiveRequest (file:///Users/bpasero/Development/Microsoft/monaco/out/vs/workbench/services/extensions/node/rpcProtocol.js:88:50)
    at RPCProtocol._receiveOneMessage (file:///Users/bpasero/Development/Microsoft/monaco/out/vs/workbench/services/extensions/node/rpcProtocol.js:71:26)
    at file:///Users/bpasero/Development/Microsoft/monaco/out/vs/workbench/services/extensions/node/rpcProtocol.js:17:92
    at file:///Users/bpasero/Development/Microsoft/monaco/out/vs/workbench/services/extensions/node/rpcProtocol.js:175:21
    at Emitter.fire (file:///Users/bpasero/Development/Microsoft/monaco/out/vs/base/common/event.js:103:38)
    at Socket.<anonymous> (file:///Users/bpasero/Development/Microsoft/monaco/out/vs/base/parts/ipc/node/ipc.net.js:90:46)
    at emitOne (events.js:116:13)
    at Socket.emit (events.js:211:7)
    at addChunk (_stream_readable.js:263:12)
    at readableAddChunk (_stream_readable.js:250:11)
    at Socket.Readable.push (_stream_readable.js:208:10)
    at Pipe.onread (net.js:594:20)
```

The first exception reproduces in stable. The second one (`mainThreadWebview` @mjbvz) seems new.

**Comment by mjbvz** [comment]
I believe the webview one was addressed by f2a2f39

**Comment by leerssej** [comment]
Sort of confirmed over here, too.  I think I recreated it even more easily though. I just have a single file.
If I save a workspace, then I trade my REPL functionality for the above warning.
If I remove the workspace file, then I can run my code again 😃
